File: models/transaction.py
from typing import Literal, TypedDict, NotRequired
from pydantic import BaseModel

TransactionType = Literal["entrada", "saida"]

class TransactionUpdate(TypedDict):
    """date -> ISO yyyy-mm-dd"""
    t_type: NotRequired[TransactionType]
    value: NotRequired[float]
    description: NotRequired[str]
    date: NotRequired[str] # ISO yyyy-mm-dd

class Registro(BaseModel):
    value: float
    description: str
    # data = { value: 2.99, description: "tubaina oxxo" }

File: frontend/src/models/Transaction.ts
export type TransactionType = "entrada" | "saida";
export type ISODate = `${number}-${number}-${number}`

export interface Transaction {
  id: number;
  value: number;
  description: string;
  /** ISO yyyy-mm-dd */
  date: ISODate;
  t_type: TransactionType;
}

export interface NewTransaction {
  value: number;
  description: string;
  /** ISO yyyy-mm-dd */
  date?: ISODate;
  t_type: TransactionType;
}

export interface Registro {
  value: number;
  description: string
}

export interface TransactionsGroup {
  entradas: Transaction[];
  saidas: Transaction[];
}

export interface TransactionsResponse {
  transacoes: TransactionsGroup;
}

export interface TransactionUpdate {
  t_type?: TransactionType;
  value?: number;
  description?: string;
  date?: ISODate;
}
